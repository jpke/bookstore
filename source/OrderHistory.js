import React from 'react';
import {connect} from 'react-redux';
import { StyleSheet,
         Text,
         TouchableHighlight,
         View,
         ScrollView
       } from 'react-native';
import formatAMPM from './dateFormatter';

class OrderHistory extends React.Component {
  constructor(props) {
    super(props)
  }
  navigate(routeName) {
    if(routeName === "HomeScreen") {
      this.props.navigator.popToTop(0);
    } else {
      this.props.navigator.push({
        name: routeName
      });
    }
  }
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <View style={{flex: 1}}>
          </View>
          <Text style={styles.title}>Order History</Text>
          <View style={{flex: 1}}></View>
        </View>
        <View style={styles.bookList}>
          <ScrollView>
            {this.props.orders.map((order, index) => {
              return (
                <View key={index}>
                  <View style={styles.rowContainer}>
                    <Text style={styles.bookTitle}>{order.title}</Text>
                    <Text style={styles.bookAuthor}>{order.author}</Text>
                    <Text style={styles.bookDate}>{"Ordered " + formatAMPM(order.date) + " on " + order.date.toDateString()}</Text>
                  </View>
                  <View style={styles.separator}/>
                </View>
                )
              })
            }
          </ScrollView>
        </View>
        <View style={styles.menu}>
          <TouchableHighlight style={styles.button} onPress={() => this.props.navigator.pop()}>
            <Text style={styles.menuItem}>Back</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => this.navigate('HomeScreen')}>
            <Text style={styles.menuItem}>Home</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    orders: state.orders
  }
}

function mapDispatchToProps(dispatch) {
  return {
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderHistory);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  header: {
    marginTop: 22,
    marginBottom: 10,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  title: {
    flex: 4,
    fontSize: 40,
    color: '#ffcd67',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4,
    textAlign: 'center'
  },
  bookList: {
    flex: 5,
    width: 320,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  rowContainer: {
    flex: 1,
    marginTop: 6,
    marginBottom: 6,
    padding: 6,
    backgroundColor: 'rgba(255,205,103,0.8)',
    borderRadius: 5,
  },
  bookTitle: {
    fontSize: 22,
    padding: 5,
    color: '#fff',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4
  },
  bookAuthor: {
    fontSize: 18,
    padding: 5,
  },
  bookDate: {
    fontSize: 15,
    padding: 5
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
  menu: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  menuItem: {
    fontSize: 18,
    color: '#FFFFFF',
  },
  button: {
    backgroundColor: '#6799FF',
    // flex: 1,
    width: 150,
    padding: 10,
    margin: 10,
    borderRadius: 10,
    alignItems: 'center',
  }
});
