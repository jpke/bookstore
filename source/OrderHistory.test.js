import React from 'react';
import ReactNative from 'react-native';
import OrderHistory from './OrderHistory';

import renderer from 'react-test-renderer';

function submitInfo(info, type, option) {
  console.log("order info to save to app state: ", info, type, option);
}

it('renders correctly', () => {
  const tree = renderer.create(
    <OrderHistory
      orders={[
        {
          title: "The Lexus and the Olive Tree",
          author: "Thomas Friedman",
          date: new Date("Sat Mar 18 2017 16:38:08 GMT-0400 (EDT)")
        },
        {
          title: "Wit and Wisdom",
          author: "Toby Reynolds",
          date: new Date("Fri Mar 17 2017 16:40:30 GMT-0400 (EDT)")
        }
      ]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
