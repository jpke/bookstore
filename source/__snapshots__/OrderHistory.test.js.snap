// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffcd67",
          "flex": 4,
          "fontSize": 40,
          "textAlign": "center",
          "textShadowColor": "black",
          "textShadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "textShadowRadius": 4,
        }
      }
    >
      Order History
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 5,
        "justifyContent": "space-around",
        "width": 320,
      }
    }
  >
    <ScrollView>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255,205,103,0.8)",
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 6,
              "marginTop": 6,
              "padding": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 22,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            The Lexus and the Olive Tree
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "padding": 5,
              }
            }
          >
            Thomas Friedman
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            Ordered 4:38 pm on Sat Mar 18 2017
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#dddddd",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255,205,103,0.8)",
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 6,
              "marginTop": 6,
              "padding": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 22,
                "padding": 5,
                "textShadowColor": "black",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "textShadowRadius": 4,
              }
            }
          >
            Wit and Wisdom
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "padding": 5,
              }
            }
          >
            Toby Reynolds
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            Ordered 4:40 pm on Fri Mar 17 2017
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#dddddd",
              "height": 1,
            }
          }
        />
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 10,
            "padding": 10,
            "width": 150,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Back
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 10,
            "padding": 10,
            "width": 150,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Home
      </Text>
    </View>
  </View>
</View>
`;
